src/components/AddressTagsPage.tsx

```diff
@@ -0,0 +1,140 @@
+import { SyncOutlined } from "@ant-design/icons";
+import { Alert, Button, Card } from "antd";
+import "antd/dist/antd.dark.css";
+import React, { useCallback, useEffect, useState } from "react";
+import { useRecords } from "../hooks/useRecords";
+import SDKSession from "../sdk/sdkSession";
+import { constants } from "../util/helpers";
+import { AddAddressesButton } from "./AddAddressesButton";
+import { AddressTable } from "./AddressTable";
+import { PageContent } from "./index";
+import { Record } from "../types/records";
+const { ADDRESSES_PER_PAGE } = constants;
+
+const AddressTagsPage = ({
+  isMobile,
+  session,
+}: {
+  isMobile: () => boolean;
+  session: SDKSession;
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [retryFunction, setRetryFunction] = useState(null);
+  const [addresses, addAddresses, removeAddresses] = useRecords([]);
+
+  const fetchRecords = useCallback(
+    (fetched = 0, retries = 1) => {
+      setIsLoading(true);
+      session
+        .getKvRecords({
+          start: fetched,
+          n: ADDRESSES_PER_PAGE,
+        })
+        .then((res) => {
+          addAddresses(res.records);
+          const totalFetched = res.fetched + fetched;
+          const remainingToFetch = res.total - totalFetched;
+          if (remainingToFetch > 0) {
+            fetchRecords(fetched + res.fetched);
```

Comment:
You need to pass `retries` too

Reply:
I think we might want to keep retries as the default there. If we pass retries  there, and let's say we are fetching 20 records: 

1. The first request fails the first try
1. `retries` is set to 0
1. The first request succeeds
1. The second request is called with 0 retries

I think we'd want the seconds request to be able to retry at least once, right?

Reply:
Correct--if it doesn't receive an arg, it will always use the default from the parameter declaration, no matter the call site.